### DISCUSSION TITLE
Helix emulator for VSCode

### DISCUSSION THREAD

**Discussion by theowenyoung** [helix-editor/helix#2975] [Ideas]
Is there any helix extension (emulator) for VSCode. So that, in command mode it will act like helix but once you enter Insert mode it will be a completely vanilla VSCode experience.

Something like -

https://marketplace.visualstudio.com/items?itemName=vscodevim.vim

Or,

https://marketplace.visualstudio.com/items?itemName=jpotterm.simple-vim

https://github.com/71/dance

https://github.com/reykjalin/kakoune-mode

https://marketplace.visualstudio.com/items?itemName=johtela.vscode-modaledit

**Comment by archseer** [comment]
Not at the moment. Converting this to a discussion since there's nothing actionable for us to resolve

**Comment by ghost** [comment]


**Reply by Ed1123** [reply]
Can you share your keybindings?

**Reply by zetashift** [reply]
> Can you share your keybindings?

Here are mine: https://gist.github.com/zetashift/22f5a66eb5cf6ca86c1fc695f7421247

**Comment by zhaob1n** [comment]
aching for helix emulator in JetBrains

**Comment by robrecord** [comment]
Here is one, but it appears not to work properly yet. A good start though? https://marketplace.visualstudio.com/items?itemName=GarlandCrow.vscode-helix

**Comment by samholmes** [comment]
If only an extension that could configure as many states and keybindings to commands as you'd like, then integrating helix/vim/kak would just be using a config file...

**Comment by anthonyjclark** [comment]
I am currently using: https://github.com/DCsunset/vscode-modal-editor

Here is my config: https://gist.github.com/anthonyjclark/1de4882c13ffd3c61863bc657557253e

It started out much more Helix-like, but it has diverged a bit to better suit my needs.

**Comment by ath3** [comment]
I think i saw this mentioned before somewhere:

https://github.com/Silverquark/dance

Its a fork of a kakoune inspired plugin with the goal to make it more helix-like.
No idea if it works properly, never used it.

**Reply by robrecord** [reply]
Yes I think this is the one - if so, it worked really well for me.

**Comment by lucyColeclough** [comment]
old and someone here may have even written this but i use this:
https://marketplace.visualstudio.com/items?itemName=silverquark.dancehelix

just for debugging :D

**Comment by robrecord** [comment]
Yes, there is a plug-in which is based on Kakoune and hence very similar to Helix's movements - it's called Dance. https://marketplace.visualstudio.com/items?itemName=gregoire.dance

EDIT I've seen others in this thread suggest Dance forks that might be more helix-like, I can't remember the correct fork since I'm not using VSCode anymore.

**Comment by jasonwilliams** [comment]
Hey all, I’ve made a helix emulator for VS Code here, it emulates most of the keybindings but there’s still a few gaps.

https://github.com/jasonwilliams/vscode-helix
